gpii/node_modules/flowManager/src/UntrustedFlowManager.js

```diff
@@ -44,16 +44,54 @@ fluid.defaults("gpii.flowManager.untrusted", {
         proximityTriggered: {
             gradeNames: "gpii.flowManager.untrusted.stateChangeHandler"
         }
+    },
+    invokers: {
+        setSettings: {
+            funcName: "gpii.flowManager.untrusted.setSettings",
+            args: [
+                "{that}.untrustedSettingsDataSource",
+                "{arguments}.0", // gpiiKey
+                "{arguments}.1", // preferences
+                "{that}.events.preferencesSavedSuccess",
+                "{that}.events.preferencesSavedError"
+            ]
+        }
+    },
+    listeners: {
+        "{lifecycleManager}.events.preferencesUpdated": {
+            namespace: "savePreferences",
+            listener: "{that}.setSettings"
+        },
+        "preferencesSavedSuccess.log": {
+            listener: "fluid.log",
+            args: ["FlowManager: save preferences to the cloud for the GPII key: ", "{arguments}.0.gpiiKey", ": ", "{arguments}.0.message"]
+        },
+        "preferencesSavedError.reportSaveError": {
+            listener: "gpii.flowManager.untrusted.reportSaveError",
+            args: ["{that}.userErrors.events.userError", "{arguments}.0"]
+        }
     }
 });
 
+gpii.flowManager.untrusted.reportSaveError = function (userErrorEvent, error) {
+    userErrorEvent.fire({
+        isError: true,
+        messageKey: "SaveToCloudFail",
+        originalError: error
+    });
+};
 
 gpii.flowManager.untrusted.getSettings = function (untrustedSettingsDataSource, gpiiKey, deviceReporterData, onMatchDone, onError) {
     var settings = untrustedSettingsDataSource.get(gpiiKey, deviceReporterData);
     settings.then(onMatchDone.fire, onError.fire);
 };
 
+gpii.flowManager.untrusted.setSettings = function (untrustedSettingsDataSource, gpiiKey, preferences, preferencesSavedSuccess, preferencesSavedError) {
+    var saveResult = untrustedSettingsDataSource.set(gpiiKey, preferences);
+    saveResult.then(preferencesSavedSuccess.fire, preferencesSavedError.fire);
```

Comment:
More useful than having the Success/Error events would be to return this promise, and also document this method